packages/i18n/src/create-i18n.js

```diff
@@ -22,13 +22,35 @@ const DEFAULT_LOCALE_DATA = {
 	},
 };
 
+/*
+ * Regular expression that matches i18n hooks like `i18n.gettext`, `i18n.ngettext`,
+ * `i18n.gettext_domain` or `i18n.ngettext_with_context`
+ */
+const I18N_HOOK_REGEXP = /^i18n\.(n?)gettext(_|$)/;
```

Comment:
Does this need to match `i18n.has_translation` as well?

Reply:
Good point 🙂 When I was adding this regexp and the code that uses it, the `i18n.has_translation` filter didn't exist yet. The filter cannot change the result of `__()`, but it can change result of `hasTranslation()` which is one of the functions that the React bindings expose. I'll add it.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/28784/commits/a245ae0c4b0171ea158d9c2fb56501f1e0fcb98d